src/extension/tools/node/findTextInFilesTool.tsx

```diff
@@ -141,11 +141,30 @@ const FIND_FILES_START_PRIORITY = 1000;
 
 export class FindTextInFilesResult extends PromptElement<FindTextInFilesResultProps> {
 	override async render(state: void, sizing: PromptSizing): Promise<PromptPiece> {
-		const textMatches = this.props.textResults.filter(isTextSearchMatch);
-		if (textMatches.length === 0) {
+		const textMatchesAll = this.props.textResults.filter(isTextSearchMatch);
+		if (textMatchesAll.length === 0) {
 			return <>No matches found</>;
 		}
-
+		// Deduplicate matches based on URI and source range
+		const textMatches: vscode.TextSearchMatch2[] = [];
```

Comment:
I think that this breaks the `maxResultsText` check?

Reply:
And re: `numResultsText` eg I wonder whether it will confuse some scenarios to report a different number of results, or maybe it doesn't matter. But now this is about "number of lines with matches" not "number of matches"